vault/expiration.go

```diff
@@ -668,22 +668,22 @@ func (m *ExpirationManager) revokePrefixCommon(prefix string, force, sync bool)
 	// Ensure there is a trailing slash; or, if there is no slash, see if there
 	// is a matching specific ID
 	if !strings.HasSuffix(prefix, "/") {
-		le, err := m.loadEntry(prefix)
+		le, err := m.loadEntry(ctx, prefix)
 		if err == nil && le != nil {
 			if sync {
-				if err := m.revokeCommon(m.quitContext, prefix, force, false); err != nil {
+				if err := m.revokeCommon(ctx, prefix, force, false); err != nil {
 					return errwrap.Wrapf(fmt.Sprintf("failed to revoke %q: {{err}}", prefix), err)
 				}
 				return nil
 			}
-			return m.LazyRevoke(prefix)
+			return m.LazyRevoke(ctx, prefix)
 		}
 		prefix = prefix + "/"
 	}
 
 	// Accumulate existing leases
 	sub := m.idView.SubView(prefix)
-	existing, err := logical.CollectKeys(m.quitContext, sub)
```

Comment:
Same deal, we've seen collectkeys take a really long time.

Reply:
Same here with the incoming context will be the active context.